src/inmanta_module_factory/inmanta/index.py

```diff
@@ -28,22 +29,32 @@ def __init__(
         self,
         path: List[str],
         entity: Entity,
-        attributes: List[Attribute],
+        *,
+        attributes: Optional[List[Attribute]] = None,
```

Comment:
Why handle them separate and not just based on a common supertype? Now you need to filter them before passing them in?

Reply:
Because they are not handled the same way in the validate method anyway (an entity doesn't have access to all its relations)

Reply:
I can accept everything in one field here, that would make it a non-breaking change, but then I have to split them based on the received type

Reply:
But then the name would be poorly chosen

Reply:
> Now you need to filter them before passing them in?

What do you mean exactly? I am only passing the attributes and relations that should be part of the index

Reply:
yes, but say that you have the names and the entity, how do you fetch them to put them here?

Reply:
also, if you define a `get_type` method on the common superclass, the check is the same for both types.....